File: app/src/main/java/hu/bencelaszlo/notes/Utils/DatabaseHandler.java
package hu.bencelaszlo.notes.Utils;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import hu.bencelaszlo.notes.Model.NoteModel;

import java.util.ArrayList;
import java.util.List;

public class DatabaseHandler extends SQLiteOpenHelper {
    private static final int VERSION = 1;
    private static final String NAME = "notesDb";
    private static final String NOTES_TABLE = "notes";
    private static final String ID = "id";
    private static final String NOTE = "note";
    private static final String CREATE_NOTE_TABLE = "CREATE TABLE " + NOTES_TABLE + "(" + ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " + NOTE + " TEXT)";

    private SQLiteDatabase db;

    public DatabaseHandler(Context context) {
        super(context, NAME, null, VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_NOTE_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // Drop older table if existed
        db.execSQL("DROP TABLE IF EXISTS " + NOTES_TABLE);
        onCreate(db);
    }

    public void openDatabase() {
        db = this.getWritableDatabase();
    }

    public void insertNote(NoteModel note){
        ContentValues cv = new ContentValues();
        cv.put(NOTE, note.getNote());
        db.insert(NOTES_TABLE, null, cv);
    }

    public List<NoteModel> getNotes(){
        List<NoteModel> notes = new ArrayList<>();
        Cursor cursor = null;
        db.beginTransaction();
        try{
            cursor = db.query(NOTES_TABLE, null, null, null, null, null, null, null);
            if(cursor != null){
                if(cursor.moveToFirst()){
                    do{
                        NoteModel note = new NoteModel();
                        note.setId(cursor.getInt(cursor.getColumnIndex(ID)));
                        note.setNote(cursor.getString(cursor.getColumnIndex(NOTE)));
                        notes.add(note);
                    }
                    while(cursor.moveToNext());
                }
            }
        }
        finally {
            db.endTransaction();
            assert cursor != null;
            cursor.close();
        }
        return notes;
    }

    public void updateNote(int id, String text) {
        ContentValues values = new ContentValues();
        values.put(NOTE, text);
        db.update(NOTES_TABLE, values, ID + "= ?", new String[] {String.valueOf(id)});
    }

    public void deleteNote(int id){
        db.delete(NOTES_TABLE, ID + "= ?", new String[] {String.valueOf(id)});
    }
}

File: app/src/main/java/hu/bencelaszlo/notes/utils/DatabaseHandler.kt
package hu.bencelaszlo.notes.utils

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import hu.bencelaszlo.notes.model.NoteModel
import java.util.ArrayList

class DatabaseHandler(context: Context?) : SQLiteOpenHelper(context, NAME, null, VERSION) {
    private lateinit var db: SQLiteDatabase

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_NOTE_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // Drop older table if existed
        db.execSQL("DROP TABLE IF EXISTS $NOTES_TABLE")
        onCreate(db)
    }

    fun openDatabase() {
        db = this.writableDatabase
    }

    fun insertNote(note: NoteModel) {
        val cv = ContentValues()
        cv.put(NOTE, note.note)
        db.insert(NOTES_TABLE, null, cv)
    }

    val notes: MutableList<NoteModel>
        @SuppressLint("Range")
        get() {
            val notes: MutableList<NoteModel> = ArrayList<NoteModel>()
            var cursor: Cursor? = null
            db.beginTransaction()
            try {
                cursor = db.query(NOTES_TABLE, null, null, null, null, null, null, null)
                if (cursor != null) {
                    if (cursor.moveToFirst()) {
                        do {
                            val note = NoteModel()
                            note.id = cursor.getInt(cursor.getColumnIndex(ID))
                            note.note = cursor.getString(cursor.getColumnIndex(NOTE))
                            notes.add(note)
                        } while (cursor.moveToNext())
                    }
                }
            } finally {
                db.endTransaction()
                assert(cursor != null)
                cursor!!.close()
            }
            return notes
        }

    fun updateNote(id: Int, text: String?) {
        val values = ContentValues()
        values.put(NOTE, text)
        db.update(NOTES_TABLE, values, "$ID= ?", arrayOf(id.toString()))
    }

    fun deleteNote(id: Int) {
        db.delete(NOTES_TABLE, "$ID= ?", arrayOf(id.toString()))
    }

    companion object {
        private const val VERSION = 1
        private const val NAME = "notesDb"
        private const val NOTES_TABLE = "notes"
        private const val ID = "id"
        private const val NOTE = "note"
        private const val CREATE_NOTE_TABLE =
            "CREATE TABLE $NOTES_TABLE($ID INTEGER PRIMARY KEY AUTOINCREMENT, $NOTE TEXT)"
    }
}